### PR TITLE
Add pre-commit hooks to automatically format code on git commit

### PR DESCRIPTION
**Pull request by jordan-gillard**
This PR adds [pre-commit](https://pre-commit.com/), a Python library for configuring and running git hooks. 

I added configuration to run the `black` code formatter automatically whenever a user runs `git commit`. It will only run on Git staged files. The pre-commit config also has checks for yaml files, adding a newline to the end of files, and removing trailing whitespaces from lines. I also ran these hooks on the entire code base, which applied minor formatting changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -105,6 +105,9 @@ pip install poetry
 # Install dependencies
 poetry install
 
+# Setup pre-commit hooks
+pre-commit install
```
Comment (GalenReich) [inline README.md:109]: Would this be better as:

`poetry run pre-commit install`

Or should we `poetry shell` before or something else?

Comment (jordan-gillard) [reply README.md:109]: Good catch! Ill take care of that in my next commit.

Comment (GalenReich) [reply README.md:109]: To double check, on the other PR?

Also, I've really appreciated this PR (and think I'll like #19 too) - I'm not a software engineer, but am always keen to learn more! These have been tremendously educational for me, so thank you 🙌

Comment (jordan-gillard) [reply README.md]: Updated the instruction here: 3d41505

There _might_ be a merge conflict in the other PR's `README.md` after this one goes in. If so it'll be a simple fix :)

>I'm not a software engineer, but am always keen to learn more! These have been tremendously educational for me, so thank you 🙌

Awww, brooo 😊 You are _absolutely_ a software engineer! 

I'm glad you find my code helpful! I'm happy to contribute and spread some knowledge.

**Review by GalenReich — APPROVED** [review]

**Pull request merged**